server/src/main/kotlin/suwayomi/tachidesk/manga/impl/download/DownloadManager.kt

```diff
@@ -92,12 +64,16 @@ object DownloadManager {
             )
             start()
         }
-        notifyAllClients()
+        val status = getStatus()
+        eventDispatcher.enqueue(status)
+        eventDispatcher.notifyAllClients(status)
     }
 
     fun unqueue(chapterIndex: Int, mangaId: Int) {
         downloadQueue.removeIf { it.mangaId == mangaId && it.chapterIndex == chapterIndex }
-        notifyAllClients()
+        val status = getStatus()
+        eventDispatcher.dequeue(status)
```

Comment:
I don't think this would ever do anything. `getStatus()` always returns new `Event` which has new `randomUUID()`. `dequeue(status)` then look for this new id for removal.

Also, I don't think this unqueue is supposed to dequeue the status. `unqueue` is used for removing specific download job. It should notify client with new status (which will be missing the unqueued download task) which it does below. I'm not sure what this code is meant to do.